compiler/modules/front/codeGen.bal

```diff
@@ -318,12 +325,19 @@ function codeGenOnPanic(CodeGenContext cx, Position pos) {
     }
 }
 
-function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Environment initialEnv, s:StmtBlock block) returns CodeGenError|StmtEffect {
+// If block of stmts doesn't complete normally, will return empty narrowings and assignments. initialNarrowings array will be modified.
+function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Environment initialEnv, s:StmtBlock block, StmtNarrowing[] initialNarrowings = []) returns CodeGenError|StmtEffect {
     bir:BasicBlock? curBlock = bb;
     Environment env = environmentCopy(initialEnv);
     final int startRegister = cx.nextRegisterNumber();
```

Comment:
This is no longer the case since I am now using `unnarrowBinding`, But I still believe this is the right place to process `initialNarrowings` since new bindings should live inside block scope. I added a comment saying that.

Reply:
Does it still modify `initialNarrowings`? If so, why?

Reply:
Yes, (but not here, in the loop) `initialNarrowings` grows into the narrowings of the block, I could clone it to keep things clean.

Reply:
Seems a bit confusing to call it initialNarrowings and then mutate it into something that isn't the initialNarrowings.

Reply:
Ya that is not nice. Easy fix is to clone. Maybe a better solution is to accept `ExprNarrowing initialNarrowing` and create the array here. But then I'd have to accept boolean as well. (Or I can get rid of the boolean, have a single `Narrowing` type that can be used for stmt as well, where `type ExprNarrowing  [Narrowing, Narrowing];`)

Reply:
I'll get around this for now by making it `StmtNarrowing? initialNarrowing` and cross this bridge when we are doing `&&` and `||`.